viz_scripts/mode_specific_timeseries.ipynb

```diff
@@ -0,0 +1,392 @@
+    "\n",
+    "data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# Count the number of trips for each confirmed mode\n",
+    "mode_counts = data.groupby(['user_id','date_time','mode_confirm'], as_index=False).size()\n",
+    "mode_counts.rename(columns={'size':'trip_count'}, inplace=True)\n",
+    "\n",
+    "# Sum daily distance traveled for each mode\n",
+    "mode_distance = data.groupby(['user_id','date_time','mode_confirm'], as_index=False)[['distance_miles']].sum()\n",
+    "mode_distance.rename(columns={'sum':'distance_miles'}, inplace=True)\n",
+    "mode_distance['distance_miles'] = mode_distance['distance_miles'].fillna(0)\n",
+    "\n",
+    "# Get the count of unique users that were active on each given date\n",
+    "active_users = pd.DataFrame(data.groupby(['date_time'], as_index=False)['user_id'].nunique())\n",
+    "active_users.rename(columns={'user_id':'active_users'}, inplace=True)\n",
+    "\n",
+    "# Add 7-day rolling avg smoothing to better see trends\n",
+    "mode_counts['trip_count_smooth'] = mode_counts.groupby(['user_id','mode_confirm'])['trip_count'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "mode_distance['distance_miles_smooth'] = mode_distance.groupby(['user_id','mode_confirm'])['distance_miles'].apply(lambda x: x.rolling(7,1).mean())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Timeseries Plots"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mode_counts_interest = mode_counts[mode_counts['mode_confirm']==mode_of_interest].copy()\n",
+    "mode_distance_interest = mode_distance[mode_distance['mode_confirm']==mode_of_interest].copy()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Mapping new mode labels with dictionaries\n",
+    "mode_counts['Mode_confirm'] = mode_counts['mode_confirm'].map(dic_re)\n",
+    "mode_counts_interest['Mode_confirm'] = mode_counts_interest['mode_confirm'].map(dic_re)\n",
+    "mode_distance_interest['Mode_confirm'] = mode_distance_interest['mode_confirm'].map(dic_re)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Daily trips for mode of interest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare trips across all users\n",
+    "plot_data = mode_counts_interest.groupby(['date_time'])['trip_count_smooth'].agg(['sum']).reset_index()\n",
+    "\n",
+    "plot_title = f'Daily {mode_of_interest} Trips\\n%s'%quality_text\n",
+    "ylab = 'Trip Count'\n",
+    "file_name = \"ts_mode_share%s\"%file_suffix\n",
```

Comment:
Not sure if this works directly since it is hard to get the notebook formatting right, but I hope you get the idea

```suggestion
    "file_name = f\"ts_{mode_of_interest}_share{file_suffix}\",
```